src/common/directives/creditCardNumber.directive.js

```diff
@@ -0,0 +1,83 @@
+import angular from 'angular'
+import appConfig from 'common/app.config'
+import cruPayments from 'cru-payments'
+
+const directiveName = 'creditCardNumber'
+
+const creditCardNumberDirective = /* @ngInject */ function () {
+  return {
+    restrict: 'A',
+    require: '?ngModel',
+    link: function (scope, element, attrs, ngModelCtrl) {
+      if (!ngModelCtrl) {
+        return
+      }
+
+      ngModelCtrl.$parsers.unshift(function (creditCardNumber) {
+        let trimmedCardNum = creditCardNumber.replace(/\s+/g, '').replace(/\D/gi, '')
```

Comment:
I believe only `creditCardNumber.replace(/\D/g, '')` is necessary to strip out all non-digits.